### PR TITLE
change: terminate Component GenServers if no one's subscribed

### PR DESCRIPTION
**Pull request by geolessel**
If no one is subscribed to a Component GenServer, it won't fetch the
api endpoint and it also won't update the `last_update` timestamp. If
there are no subscribers, and it's been more that 10 minutes since the
last listened-to update, terminate the GenServer. We can easily spin
it back up again when needed.

Satisfies #25.

### CODE REVIEW

File: apps/dashboard/lib/dashboard/component.ex [hunk 1]
```diff
@@ -1,19 +1,23 @@
 defmodule Dashboard.Component do
   defmacro __using__(_opts) do
     quote do
-      use GenServer
+      use GenServer, restart: :transient
```
Comment (geolessel) [inline apps/dashboard/lib/dashboard/component.ex:4]: Before I added this, the GenServer would shut down and then I'd start to see API requests again. I thought to myself, "it's like something in the code is restarting the server once it terminates." In fact, it was doing exactly what a good Supervisor _should_ do! 😆 

The default is `restart: :permanent`, which will always restart it. `restart: :transient` will only restart it if its shutdown was abnormal.

Comment (jessejanderson) [reply apps/dashboard/lib/dashboard/component.ex:4]: https://www.youtube.com/watch?v=4SCwubzqsVU

**Comment by geolessel** [comment]
That made me LOL.

File: apps/dashboard/lib/dashboard/component.ex (hunk 1 above)
Comment (geolessel) [reply apps/dashboard/lib/dashboard/component.ex:4]: That made me LOL

**Pull request merged**